app/jobs/importing/hud_zip/fetch_and_import_job.rb

```diff
@@ -7,22 +7,46 @@
 module Importing::HudZip
   class FetchAndImportJob < BaseJob
     queue_as ENV.fetch('DJ_LONG_QUEUE_NAME', :long_running)
+    WAIT_MINUTES = 15
+
+    # When you call jobs with .perform_later, they are executed in the ActiveJob world, which doesn't
+    # obey they max_attempts for Delayed Job.  We'll adjust the attempts to give us what we want
+    after_enqueue :enforce_max_attempts
 
     def perform(klass:, options:)
       safe_klass = known_classes.detect { |m| klass == m }
-      raise "Unknown import class: #{klass}; You must add it to the whitelist in FetchAndImportJob" unless safe_klass.present?
+      raise "Unknown import class: #{klass}; You must add it to the list of known classes in FetchAndImportJob" unless safe_klass.present?
 
-      safe_klass.constantize.new(**options).import!
-    end
+      data_source_id = options[:data_source_id]
+      lock_obtained = GrdaWarehouse::DataSource.with_advisory_lock(advisory_lock_name(data_source_id), timeout_seconds: 60) do
+        safe_klass.constantize.new(**options).import!
+        # To prevent re-running when called against the same files if run more than once in a day, yield true
+        true
+      end
 
-    def max_attempts
-      1
+      requeue_at(Time.current + WAIT_MINUTES.minutes, "Import of Data Source: #{data_source_id} already running...re-queuing job for #{WAIT_MINUTES} minutes from now") unless lock_obtained
     end
 
     def known_classes
       [
         'Importers::HmisAutoMigrate::S3',
       ].freeze
     end
+
+    private def advisory_lock_name(data_source_id)
+      GrdaWarehouse::DataSource.import_advisory_lock_name(data_source_id)
+    end
+
+    def enforce_max_attempts
+      delayed_job.update!(attempts: calculated_attempts)
+    end
+
+    def calculated_attempts
+      dj_max_attempts = Delayed::Worker.max_attempts
+      max_attempts = 1
+      return 0 if max_attempts >= dj_max_attempts
+
+      dj_max_attempts - max_attempts
+    end
```

Comment:
So this sets the job attempts to 1 less than the max. The intention is that it should always re queue?

Also since the return would be constant, maybe it can be written a little more concisely
```suggestion
    def calculated_attempts
      [0, Delayed::Worker.max_attempts - 1].max
    end
```

Reply:
The intention here is to limit the number of times Delayed Job attempts to run this job if it fails.  With the way we're queuing it now with `.perform_later` there's no way to indicate how many times the job should be attempted.  To work around that, we just pretend like it's already failed all but one time so the next time it fails DelayedJob/ActiveJob marks it as fully failed. 

Re-queueing is unrelated to failures, and should only happen if the lock is unavailable.